### ISSUE TITLE
[Feature Request] Autostart batch file

### ISSUE THREAD

**Issue by pukkandan** [chrisant996/clink#136]
I want a batch file to be loaded when an interactive command prompt opens (not when prompt shows). It first seems like this is already possible, but none of the ways I attempted to give me what I want

Here's what I tried:
1. Putting the batchfile in `autorun` (either alongside clink, or running clink from it) - When an `AutoRun` batch file calls another `.bat` file or simply use `call` to jump to a label, the console crashes. My script can't avoid having a `call`.
2. `clink.autostart` - This only loads the file when the prompt appears. So if I run `cmd /k some_cmd`, the the batch file only runs after `some_cmd`, but I need it to run before it. The most important use-case of this is to make variables created by the autostart file available in some_cmd
3. Using `os.exec` or `io.popen` in lua - This starts the batch file in a sperate process making the doskeys and variables not available in the main process. 

So in summary, I am requesting for a way to run script in autorun:
1. With full scripting capabilities enabled (aka, `call`)
1. Before any command execution given by `/k` AND
2. In the same process so that the environment is shared

Also, related, is there a way to run a batch file in LUA without creating a new process? 🤔 If this is possible, then it should satisfy my needs

**Comment by pukkandan** [comment]
> It sounds like the script in question has a bug. A common mistake is to accidentally invoke additional cmd.exe processes from within the script -- that would start cmd, autorun the script which starts another cmd, which autoruns the script and starts another cmd, which autoruns the script and starts another cmd, in an infinite cycle.Sorry for the extremely delayed reply. You were correct

**Comment by pukkandan** [comment]
> It sounds like the script in question has a bug. A common mistake is to accidentally invoke additional cmd.exe processes from within the script -- that would start cmd, autorun the script which starts another cmd, which autoruns the script and starts another cmd, which autoruns the script and starts another cmd, in an infinite cycle.

Sorry for the extremely delayed reply. But you were right about this. I had a for loop in the script. I thought I was being careful by running `cmd /d` inside the for loop, but totally forgot that the for loop itself starts a cmd first
